File: packages/react-native-nitro-test/ios/HybridSomeExternalObjectSubclass.swift
//
//  SomeExternalObjectSubclass.swift
//  react-native-nitro-test
//
//  Created by Marc Rousavy on 30.10.25.
//

import Foundation

class HybridSomeExternalObjectSubclass: HybridSomeExternalObjectSubclassSpec {
  func getValue() throws -> String {
    return "overridden!"
  }

  func getSubclassedValue() -> String {
    return "subclassed!"
  }
}

File: packages/react-native-nitro-test/android/src/main/java/com/margelo/nitro/test/HybridSomeExternalObjectSubclass.kt
package com.margelo.nitro.test

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridSomeExternalObjectSubclass : HybridSomeExternalObjectSubclassSpec() {
  override fun getValue(): String {
    return "overridden!"
  }

  override fun getSubclassedValue(): String {
    return "subclassed!"
  }
}
